testsuite/tests/prometheus/apicast/selfmanaged/conftest.py

```diff
@@ -0,0 +1,46 @@
+"""Provide custom gateway for tests changing apicast parameters."""
+
+from weakget import weakget
+import pytest
+
+from testsuite.gateways import gateway
+from testsuite.gateways.apicast.template import TemplateApicast
+from testsuite.utils import blame, warn_and_skip
+
+
+@pytest.fixture(scope="module", autouse=True)
+def require_openshift(testconfig):
+    """These tests require openshift available"""
+    if not weakget(testconfig)["openshift"]["servers"]["default"] % False:
+        warn_and_skip("All monitoring for selfmanaged apicast skipped due to missing openshift")
+
+
+@pytest.fixture(scope="module")
+def staging_gateway(request, gateway_kind, gateway_environment, gateway_options):
+    """Deploy self-managed template based apicast gateway."""
+    gw = gateway(kind=gateway_kind, staging=True, name=blame(request, "gw"), **gateway_options)
```

Comment:
If you are using one gateway for all tests in this directory you can simply specify it here

Reply:
No, I mean getting rid of the gateway kind fixtures all together and using OperatorApicast as a kind instead